pulsar/consumer_impl.go

```diff
@@ -575,15 +575,21 @@ func (c *consumer) Seek(msgID MessageID) error {
 	c.Lock()
 	defer c.Unlock()
 
-	if len(c.consumers) > 1 {
-		return newError(SeekFailed, "for partition topic, seek command should perform on the individual partitions")
-	}
-
 	mid, ok := c.messageID(msgID)
 	if !ok {
 		return nil
 	}
 
+	if mid.partitionIdx < 0 {
+		return newError(SeekFailed, "partitionIdx is negative")
+	}
```

Comment:
Will this break the non-partitioned topic message acknowledgment?
The partition index -1 means the message from a non-partitioned topic.

@freeznet @wolfstudy Please help confirm

Reply:
Is that really true? The pre-existing code would do the following:
```
return c.consumers[mid.partitionIdx].Seek(mid)
```

This suggests that the partition index is 0 for non-partitioned topics. Otherwise seek() would've always resulted in an index out of bounds panic.

Reply:
In Go SDK, the partition index less than 0 is not allowed.

```
// did we receive a valid partition index?
	if partition < 0 || partition >= len(c.consumers) {
		c.log.Warnf("invalid partition index %d expected a partition between [0-%d]",
			partition, len(c.consumers))
		return trackingMessageID{}, false
	}
```

And why do we need to modify the current logic, where the number of consumers is equivalent to the number of partitions, and each partition corresponds to a unique consumer.

The Go SDK currently does not support seek operations on partitioned topics, and this logic is still being implemented.

Reply:
Thanks for the comments. I have a few questions.

> And why do we need to modify the current logic, where the number of consumers is equivalent to the number of partitions, and each partition corresponds to a unique consumer.
I don't understand what you mean. I'm not changing the current logic. I'm just passing through the message id to the partitionConsumer responsible for the partition.

> The Go SDK currently does not support seek operations on partitioned topics, and this logic is still being implemented.
All right. Do you know what work still remains to have the Go client support seeking by partition?